### PR TITLE
Positional and Optional Arguments (Ref: #11018)

### PR DESCRIPTION
**Pull request by vijay-jaisankar**
Relevant issues and PRs: #11018 #10942

In this PR, the following functionalities have been added:
- Usage of `OptionParser` for positional and optional arguments 
- Special care has been taken not to use `f`strings (to maintain compatibility with python 2)

### CODE REVIEW

**Comment by behrisch** [comment]
Please do not move the call to argument parsing into the main function. This makes it harder to reuse the scripts functionality as a module. But moving the parsing into a separate function is fine (although sometimes maybe a bit overkill)

**Comment by vijay-jaisankar** [comment]
Sure thing, thanks for the feedback.

Please check if this new commit is okay, the `parse_args` method has been moved out of the `main` functions.

**Comment by behrisch** [comment]
Yes, thank you. Another point is the USAGE string. You do not use it so there is no point in defining it.

File: tools/output/vehrouteDiff.py [hunk 1]
```diff
@@ -25,14 +25,26 @@
...
 
 
+def parse_args():
+    USAGE = "Usage: " + sys.argv[0] + " <orig> <new> <out> [--earliest]"
+    optParser = ArgumentParser()
+    optParser.add_argument("orig", help="Original XML File")
+    optParser.add_argument("new", help="New XML File")
+    optParser.add_argument("out", help = "Output File")
+    optParser.add_option("--earliest", action="store_true",
+                         default=None, help="Earliest Output Parameter (Default: None)")
```
Comment (behrisch) [inline tools/output/vehrouteDiff.py:38]: the default should be False here

**Comment by vijay-jaisankar** [comment]
Sure thing, in the most recent commit, I have removed those lines and changed the default value you mentioned in the PR feedback

File: tools/output/vehrouteDiff.py (hunk 1 above)
Comment (vijay-jaisankar) [reply tools/output/vehrouteDiff.py]: Updated here: Commit 68a8cd4be1837f0a1d27d853da6ccabed4628210

**Comment by behrisch** [comment]
Thanks a lot! There are some minor style issues but I will fix them myself.

**Pull request merged**